### PR TITLE
Do two equal time steps instead of one big one small

### PR DESCRIPTION
**Pull request by paveltomin**
Relevant to time step cut situations where chopping and increasing time step can lead to a case when `nextDt` is close to `dtRemaining` but an epsilon smaller leading to a tiny last time step that on top of that later used for next time step value

Observed issue in certain situations when running spe11b

Updated baselines: GEOS-DEV/integratedTests/pull/95

### CODE REVIEW

**Review by dkachuma — COMMENTED** [review]
Looks good to me.

**Review by dkachuma — APPROVED** [review]

File: src/coreComponents/physicsSolvers/SolverBase.cpp [hunk 1]
```diff
@@ -268,7 +268,16 @@ bool SolverBase::execute( real64 const time_n,
     if( dtRemaining > 0.0 )
     {
       nextDt = setNextDt( dtAccepted, domain );
-      nextDt = std::min( nextDt, dtRemaining );
+      if( nextDt < dtRemaining )
+      {
+        // better to do two equal steps than one big and one small (even potentially tiny)
+        if( nextDt * 2 > dtRemaining )
```
Comment (jafranc) [inline src/coreComponents/physicsSolvers/SolverBase.cpp:274]: Shouldn't we be even stronger like 
```suggestion
        if( nextDt  > .7*dtRemaining )
```
?

**Review by jafranc — APPROVED** [review]

File: src/coreComponents/physicsSolvers/SolverBase.cpp (hunk 1 above)
Comment (paveltomin) [reply src/coreComponents/physicsSolvers/SolverBase.cpp:274]: could you please clarify?

Comment (jafranc) [reply src/coreComponents/physicsSolvers/SolverBase.cpp:274]: Neverming you're spliting if `nextDt` is above .5*`dtRemaining` so you're already at most conservative threshold.

File: inputFiles/poromechanicsFractures/ExponentialDecayPermeability_conformingFracture_base.xml [hunk 2]
```diff
@@ -51,8 +51,7 @@
       targetRegions="{ Domain, Fracture }">
       <NonlinearSolverParameters
         newtonTol="1.0e-2"
-        newtonMaxIter="8"
-        allowNonConverged="1"/>
```
Comment (paveltomin) [inline inputFiles/poromechanicsFractures/ExponentialDecayPermeability_conformingFracture_base.xml:55]: not needed and confusing

File: inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_faultSlip_base.xml [hunk 3]
```diff
@@ -52,8 +52,7 @@
       targetRegions="{ Domain, Fracture }">
       <NonlinearSolverParameters
         newtonTol="1.0e-2"
-        newtonMaxIter="8"
-        allowNonConverged="1"/>
```
Comment (paveltomin) [inline inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_faultSlip_base.xml:56]: not needed and confusing

File: inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_base.xml [hunk 4]
```diff
@@ -51,8 +51,7 @@
       targetRegions="{ Domain, Fracture }">
       <NonlinearSolverParameters
         newtonTol="1.0e-2"
-        newtonMaxIter="8"
-        allowNonConverged="1"/>
```
Comment (paveltomin) [inline inputFiles/poromechanicsFractures/PoroElastic_conformingFracture_2d_openingFrac_base.xml:55]: not needed and confusing

File: inputFiles/singlePhaseFlowFractures/fractureFlow_conforming_2d_vtk_input.xml [hunk 5]
```diff
@@ -10,8 +10,7 @@
       targetRegions="{ Region, Fracture }">
     <NonlinearSolverParameters
       newtonTol="1.0e-2"
-      newtonMaxIter="8"
-      allowNonConverged="1"/>
```
Comment (paveltomin) [inline inputFiles/singlePhaseFlowFractures/fractureFlow_conforming_2d_vtk_input.xml:14]: not needed and confusing

File: inputFiles/singlePhaseFlowFractures/fractureMatrixFlow_conforming_2d.xml [hunk 6]
```diff
@@ -10,8 +10,7 @@
       targetRegions="{ Region2, Fracture }">
       <NonlinearSolverParameters
         newtonTol="1.0e-2"
-        newtonMaxIter="8"
-        allowNonConverged="1"/>
```
Comment (paveltomin) [inline inputFiles/singlePhaseFlowFractures/fractureMatrixFlow_conforming_2d.xml:14]: not needed and confusing

File: inputFiles/singlePhaseFlowFractures/impermeableFault_conforming_base.xml [hunk 7]
```diff
@@ -10,8 +10,7 @@
       targetRegions="{ RockMatrix, Fracture }">
       <NonlinearSolverParameters
         newtonTol="1.0e-2"
-        newtonMaxIter="8"
-        allowNonConverged="1"/>
```
Comment (paveltomin) [inline inputFiles/singlePhaseFlowFractures/impermeableFault_conforming_base.xml:14]: not needed and confusing

File: inputFiles/thermalSinglePhaseFlowFractures/fractureMatrixThermalFlow_conforming_base.xml [hunk 8]
```diff
@@ -17,7 +17,7 @@
       <NonlinearSolverParameters
         newtonTol="1.0e-2"
         newtonMaxIter="8"
-        allowNonConverged="1"/>
```
Comment (paveltomin) [inline inputFiles/thermalSinglePhaseFlowFractures/fractureMatrixThermalFlow_conforming_base.xml:20]: runs just fine after modification

**Pull request merged**